src/diffusers/models/vae.py

```diff
@@ -391,7 +395,7 @@ def forward(self, sample: torch.FloatTensor) -> torch.FloatTensor:
         return dec
 
 
-class AutoencoderKL(ModelMixin, ConfigMixin):
+class AutoencoderKL(ModelMixin, ConfigMixin, MPSWarmupMixin):
```

Comment:
Not a big fan of an additional abstraction - I'd like to avoid it here first because of the following:

- Makes the code harder to understand for everybody that's not using MPS
- Makes MPS somewhat hard to debug -> now when people code just one forward pass, actually two forward passes are done
- Too much black magic - I'd prefer to instead make sure our docs are good here instead